### ISSUE TITLE
Constraining Eager Loads using Include

### ISSUE THREAD

**Issue by jonagoldman** [dingo/api#735]
Includes are auto eager loaded, and that's great, but how can I constrain the eager load?

For example, lets say an API user does something like ```GET /api/reservations?include=date```
and the 'date' relationship has 1 million rows. I want to constrain the eager load of the 'date' relationship to the latest 30 rows. Or even better, I want to pass the number of rows in a parameter.

I know its possible to request ```?include=date:limit(30)```  and then modify the results inside the transformer by reading the parameters passed to Fractal's ```ParamBag```.
But that will be executed after the actual query to the DB, and will not really constrain the eager load.

Basically I want to use the automatic include of DingoAPI and custom constrains.
I guess I should create a Transformation Layer and implement the functionality myself, but would like to hear you opinions.

Thanks.

**Comment by jasonlewis** [comment]
Yeah nothing built in for this. You'll need to configure something yourself for that level of control.

If it were me I probably wouldn't run includes for that sort of stuff. I tend to only use includes when there's a limited amount of data being included. I see that data as supporting data and not really crucial to the request.

For larger/complicated data sets I'll stick to a separate request where I can have finer control over how the request is run and how the data is returned.

That's my personal view on it and something I've been working on on an existing API (that uses this package) that deals with a LOT of linked data.

**Comment by jonagoldman** [comment]
I agree that includes are only supporting data and not crucial to the request.

But, here's the problem:
I eager load some relationship ```->with(['posts'])```, here I can constrain it to lets say the latest 30 posts. Then I specify the 'posts' relationship inside the transformer ```$availableIncludes = ['posts']``` in order to tell Fractal to transform that relationship.

But as a side effect, now anyone can add '?include' to the path and they will get all the posts (no constrains), and I don't have control over it. If I don't set the include, then I cant transform the relationship at all...

I hope the problem is clear.
